### ISSUE TITLE
Patching an Assembly with Property crashes the compiler

### ISSUE THREAD

**Issue by LinuxDoku** [LinuxDoku/ResiLab.AssemblyPatcher#9]
When patching an assembly with a Property, it generates this generic backing property - which causes roslyn compiler exceptions:

```
System.Exception : Could not compile C# method body 
namespace HelloWorld {
    public class Program {
        private static System.String _helloWorld = default(System.String);
        private static System.String <HelloWorldProperty>k__BackingField = default(System.String);

        public  static System.String get_HelloWorldProperty()
       {
            return default(System.String);
        }

        public  static void set_HelloWorldProperty(System.String value)
       {
        }

        public  static System.String GenerateHelloWorld()
       {
            return default(System.String);
        }

        public  static void Main(System.String[] args)
       {
            System.Console.WriteLine("This is a patched Hello World!");
        }
    }
}

. Errors: (4,31): error CS0308: The non-generic type 'string' cannot be used with type arguments
(4,39): error CS0246: The type or namespace name 'HelloWorldProperty' could not be found (are you missing a using directive or an assembly reference?)
(3,38): warning CS0414: The field 'Program._helloWorld' is assigned but its value is never used
   bei ResiLab.AssemblyPatcher.CodeGenerator.Compiler.MethodCodeCompiler.Compile(String cSharpMethodBodyCode, MethodDefinition methodDefinition)
```

**Comment by LinuxDoku** [comment]
Update: CSharpCodeGenerator has to ignore members with the `System.Runtime.CompilerServices.CompilerGeneratedAttribute`

**Comment by LinuxDoku** [comment]
Fixed with commit 990a69fd111c401faaa186c8f49bc49ecbce9c91
